ui-src/src/components/HoverOverlay.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+
+import layoutFormula from '../drawing/layoutFormula'
+import {
+    goalWidth
+} from '../drawing/dimensions'
+import {
+    unselectAll
+} from '../selection/actions'
+import {
+    openGoalForm,
+    updateContent
+} from '../goal-form/actions'
+
+function HoverOverlay(props) {
+    const { hoveredAddress, goalCoord, goalContent, onEditClick, onExpandClick } = props
+    const { x, y } = props.position
+    return (
+        <ul className='hover_overlay' style={{ top: `${y}px`, left: `${x}px` }}>
+            {/* <li onClick={() => onExpandClick(hoveredAddress)}>Expand</li> */}
+            <li onClick={() => onEditClick(hoveredAddress, goalCoord, goalContent)}>Edit</li>
+        </ul>
+    )
+}
+
+HoverOverlay.propTypes = {
+    hoveredAddress: PropTypes.string.isRequired,
+    position: PropTypes.shape({
+        x: PropTypes.number,
+        y: PropTypes.number
+    }).isRequired,
+    goalCoord: PropTypes.shape({
+        x: PropTypes.number,
+        y: PropTypes.number
+    }).isRequired,
+    goalContent: PropTypes.string.isRequired,
+    onEditClick: PropTypes.func.isRequired,
+    onExpandClick: PropTypes.func.isRequired
+}
+
+function mapStateToProps(state) {
+    const hoveredAddress = state.ui.hover.hoveredGoal // null or an address
+    let position, goalCoord, goalContent
+    if (hoveredAddress) {
+        goalCoord = layoutFormula(state.ui.screensize.width, state.goals, state.edges)[hoveredAddress]
+        position = {
+            x: goalCoord.x + goalWidth - 10 - 77, // 77 is the element size, this is the wrong way to do this
```

Comment:
this is here because I was spending too long fiddling around trying to make the layout work the right way. For some reason this simple CSS problem stumped me...